sdks/go/pkg/beam/io/textio/textio.go

```diff
@@ -93,11 +96,18 @@ func readFn(ctx context.Context, filename string, emit func(string)) error {
 	}
 	defer fd.Close()
 
-	scanner := bufio.NewScanner(fd)
-	for scanner.Scan() {
-		emit(scanner.Text())
+	rd := bufio.NewReader(fd)
+	for {
+		line, err := rd.ReadString('\n')
```

Comment:
One thing that caught my eye on the ReadString description was that the string returned includes the delimiter, which differs from the old scanner implementation which strips the delimiter. To keep parity between the old and new behavior you'll probably want to strip the newline before emitting the string.

Reply:
ack. done.